### PR TITLE
[WIP] Add events endpoint to DDF params_for_create

### PR DESCRIPTION
**Pull request by agrare**
Part of ManageIQ/manageiq#18818

### CODE REVIEW

**Comment by agrare** [comment]
cc @skateman

**Comment by skateman** [comment]
@agrare wow, you learned DDF? :laughing:

File: app/models/manageiq/providers/nuage/manager_mixin.rb [hunk 1]
```diff
@@ -7,42 +7,138 @@ def params_for_create
         :title  => "Configure Nuage",
```
Comment (skateman) [inline app/models/manageiq/providers/nuage/manager_mixin.rb]: This line is not needed

File: app/models/manageiq/providers/nuage/manager_mixin.rb [hunk 2]
```diff
@@ -7,42 +7,138 @@ def params_for_create
...
+            :fields    => [
+              {
+                :component => 'tab-item',
+                :name      => 'default',
+                :title     => _('Default'),
+                :fields    => [
+                  {
+                    :component              => 'validate-provider-credentials',
+                    :name                   => 'endpoints.default.valid',
+                    :validationDependencies => %w[name],
```
Comment (skateman) [inline app/models/manageiq/providers/nuage/manager_mixin.rb]: `type` is a must have here and I don't think `name` is needed

File: app/models/manageiq/providers/nuage/manager_mixin.rb [hunk 3]
```diff
@@ -7,42 +7,138 @@ def params_for_create
...
+                        :name         => "endpoints.default.default_api_port",
+                        :label        => _("API Port"),
+                        :type         => "number",
+                        :initialValue => 443,
+                        :isRequired   => true,
+                        :validate     => [{:type => "required-validator"}],
+                      },
+                      {
+                        :component  => "text-field",
+                        :name       => "endpoints.default.default_userid",
```
Comment (skateman) [inline app/models/manageiq/providers/nuage/manager_mixin.rb]:
```suggestion
                        :name       => "authentications.default.userid",
```

File: app/models/manageiq/providers/nuage/manager_mixin.rb [hunk 4]
```diff
@@ -7,42 +7,138 @@ def params_for_create
...
+                      {
+                        :component  => "text-field",
+                        :name       => "endpoints.default.default_hostname",
+                        :label      => _("Hostname (or IPv4 or IPv6 address)"),
+                        :isRequired => true,
+                        :validate   => [{:type => "required-validator"}],
+                      },
+                      {
+                        :component    => "text-field",
+                        :name         => "endpoints.default.default_api_port",
```
Comment (skateman) [inline app/models/manageiq/providers/nuage/manager_mixin.rb]:
```suggestion
                        :name         => "endpoints.default.port",
```

File: app/models/manageiq/providers/nuage/manager_mixin.rb [hunk 5]
```diff
@@ -7,42 +7,138 @@ def params_for_create
...
+                      {
+                        :component  => "text-field",
+                        :name       => "endpoints.default.default_userid",
+                        :label      => "Username",
+                        :isRequired => true,
+                        :validate   => [{:type => "required-validator"}],
+                      },
+                      {
+                        :component  => "text-field",
+                        :name       => "endpoints.default.password",
```
Comment (skateman) [inline app/models/manageiq/providers/nuage/manager_mixin.rb]:
```suggestion
                        :name       => "authentications.default.password",
```

File: app/models/manageiq/providers/nuage/manager_mixin.rb [hunk 6]
```diff
@@ -7,42 +7,138 @@ def params_for_create
...
+                ]
+              },
+              {
+                :component => 'tab-item',
+                :name      => 'events',
+                :title     => _('Events'),
+                :fields    => [
+                  {
+                    :component    => 'select-field',
+                    :name         => 'event_stream_selection',
```
Comment (skateman) [inline app/models/manageiq/providers/nuage/manager_mixin.rb]: This should be adjusted after we have the schema changes, this `event_stream_selection` doesn't exist.

File: app/models/manageiq/providers/nuage/manager_mixin.rb [hunk 7]
```diff
@@ -7,42 +7,138 @@ def params_for_create
...
+                    :initialValue => 'ceilometer',
```
Comment (skateman) [inline app/models/manageiq/providers/nuage/manager_mixin.rb]: You probably don't want this line :laughing:

File: app/models/manageiq/providers/nuage/manager_mixin.rb [hunk 8]
```diff
@@ -7,42 +7,138 @@ def params_for_create
...
+                      {
+                        :component  => "text-field",
+                        :name       => "endpoints.amqp.amqp_hostname",
+                        :label      => _("Hostname (or IPv4 or IPv6 address)"),
+                        :isRequired => true,
+                        :validate   => [{:type => "required-validator"}],
+                      },
+                      {
+                        :component    => "text-field",
+                        :name         => "endpoints.amqp.amqp_api_port",
```
Comment (skateman) [inline app/models/manageiq/providers/nuage/manager_mixin.rb]:
```suggestion
                        :name         => "endpoints.amqp.port",
```

File: app/models/manageiq/providers/nuage/manager_mixin.rb [hunk 9]
```diff
@@ -7,42 +7,138 @@ def params_for_create
...
+                    :name                   => 'endpoints.amqp.valid',
+                    :validationDependencies => %w[type event_stream_selection],
+                    :condition              => {
+                      :when => 'event_stream_selection',
+                      :is   => 'amqp',
+                    },
+                    :fields                 => [
+                      {
+                        :component  => "text-field",
+                        :name       => "endpoints.amqp.amqp_hostname",
```
Comment (skateman) [inline app/models/manageiq/providers/nuage/manager_mixin.rb]:
```suggestion
                        :name       => "endpoints.amqp.hostname",
```

File: app/models/manageiq/providers/nuage/manager_mixin.rb [hunk 10]
```diff
@@ -7,42 +7,138 @@ def params_for_create
...
+                        :name         => "endpoints.amqp.amqp_api_port",
+                        :label        => _("API Port"),
+                        :type         => "number",
+                        :isRequired   => true,
+                        :initialValue => 5672,
+                        :validate     => [{:type => "required-validator"}],
+                      },
+                      {
+                        :component  => "text-field",
+                        :name       => "endpoints.amqp.amqp_userid",
```
Comment (skateman) [inline app/models/manageiq/providers/nuage/manager_mixin.rb]:
```suggestion
                        :name       => "authentications.amqp.userid",
```

File: app/models/manageiq/providers/nuage/manager_mixin.rb [hunk 11]
```diff
@@ -7,42 +7,138 @@ def params_for_create
...
+                        :label      => "Username",
+                        :isRequired => true,
+                        :validate   => [{:type => "required-validator"}],
+                      },
+                      {
+                        :component  => "text-field",
+                        :name       => "endpoints.amqp.password",
```
Comment (skateman) [inline app/models/manageiq/providers/nuage/manager_mixin.rb]:
```suggestion
                        :name       => "authentications.amqp.password",
```

**Comment by agrare** [comment]
>  wow, you learned DDF

I can copy&paste with the best of 'em ;)

File: app/models/manageiq/providers/nuage/manager_mixin.rb [hunk 12]
```diff
@@ -51,20 +146,28 @@ def params_for_create
...
-    #   }
+    #     },
+    #   },
+    # }
     def verify_credentials(args)
       default_endpoint = args.dig("endpoints", "default")
 
       username, password, server, port, protocol = default_endpoint&.values_at(
-        "username", "password", "server", "port", "protocol"
+        "default_userid", "password", "default_hostname", "default_api_port", "default_security_protocol"
```
Comment (skateman) [inline app/models/manageiq/providers/nuage/manager_mixin.rb]: This probably won't work anymore, but take a look at how I'm doing it in Amazon for inspiration

Comment (agrare) [reply app/models/manageiq/providers/nuage/manager_mixin.rb]: Yeah I need to update this

File: app/models/manageiq/providers/nuage/manager_mixin.rb [hunk 13]
```diff
@@ -51,20 +146,28 @@ def params_for_create
...
     #   "endpoints" => {
     #     "default" => {
-    #       "username" => String,
+    #       "default_userid" => String,
+    #       "default_hostname" => String,
+    #       "default_api_port" => Integer,
+    #       "default_security_protocol" => String,
+    #       "password" => String,
+    #     },
+    #     "amqp" => {
```
Comment (Fryguy) [inline app/models/manageiq/providers/nuage/manager_mixin.rb]: I was expecting we would accept "events" as the key and under the covers we'd handle putting it in the right columns...is that not right?

Comment (skateman) [reply app/models/manageiq/providers/nuage/manager_mixin.rb]: Yeah, we're waiting for the schema change and then we can adjust this :wink:

File: app/models/manageiq/providers/nuage/manager_mixin.rb [hunk 14]
```diff
@@ -4,45 +4,140 @@ module ManageIQ::Providers::Nuage::ManagerMixin
...
+                :component => 'tab-item',
+                :name      => 'events',
+                :title     => _('Events'),
+                :fields    => [
+                  {
+                    :component    => 'select-field',
+                    :name         => 'event_stream_selection',
+                    :initialValue => 'amqp',
+                    :label        => _('Type'),
+                    :options      => [
```
Comment (Fryguy) [inline app/models/manageiq/providers/nuage/manager_mixin.rb]: I think this need `None` as an option and then it should be the default

Comment (agrare) [reply app/models/manageiq/providers/nuage/manager_mixin.rb]: :+1: added

Comment (Fryguy) [reply app/models/manageiq/providers/nuage/manager_mixin.rb]: @skateman Is there a more general way to say "event creds are not required"?

File: app/models/manageiq/providers/nuage/manager_mixin.rb [hunk 15]
```diff
@@ -4,45 +4,144 @@ module ManageIQ::Providers::Nuage::ManagerMixin
...
+                      {
+                        :label => _("AMQP"),
+                        :value => "amqp"
+                      },
+                      {
+                        :label => _("None"),
+                        :value => nil
```
Comment (Fryguy) [inline app/models/manageiq/providers/nuage/manager_mixin.rb]: None should probably be first and also the initial value?

Comment (skateman) [reply app/models/manageiq/providers/nuage/manager_mixin.rb]: yeah, you can just delete this line

File: app/models/manageiq/providers/nuage/manager_mixin.rb (hunk 14 above)
Comment (skateman) [reply app/models/manageiq/providers/nuage/manager_mixin.rb]: Not really, just have the `none` with an empty string as the value

**Comment by skateman** [comment]
I guess I should do this based on my taste :laughing:

**Pull request closed without merging**